src/status_im2/contexts/communities/overview/view.cljs

```diff
@@ -88,170 +87,121 @@
                        :margin-top    10
                        :margin-bottom 8}
                       (map-indexed (fn [inner-index channel-data] [rn/view {:key (str inner-index (:name channel-data)) :margin-top 4}
-                                                                   [quo/channel-list-item channel-data]]) (first-category list-of-channels))]]))
-                list-of-channels)])
-
-(defn icon-color []
-  (colors/theme-colors
-   colors/white-opa-40
-   colors/neutral-80-opa-40))
-
-(defn get-platform-value [value] (if platform/ios? (+ value 44) value))
-
-(def scroll-0 (if platform/ios? -44 0))
-(def scroll0 (if platform/ios? 44 0))
-(def scroll1 (if platform/ios? 86 134))
-
-(def scroll2 (if platform/ios? -26 18))
-
-(def max-image-size 80)
-(def min-image-size 32)
-
-(defn diff-with-max-min [value maximum minimum]
-  (->>
-   (+ value scroll0)
-   (- maximum)
-   (max minimum)
-   (min maximum)))
-
-(defn get-header-size [scroll-height]
-  (if (<= scroll-height scroll2)
-    0
-    (->>
-     (+ (get-platform-value -17) scroll-height)
-     (* (if platform/ios? 3 1))
-     (max 0)
-     (min (if platform/ios? 100 124)))))
-
-(defn community-card-page-view [{:keys [name description locked joined
-                                        status tokens cover tags community-color] :as community}]
-  (let [community-icon (memoize (fn [] [communities.icon/community-icon-redesign community 24]))
-        scroll-height (reagent/atom scroll-0)
-        channel-heights (reagent/atom [])
-        first-channel-height (reagent/atom 0)]
-
+                                                                   [quo/channel-list-item channel-data]]) (first-category mock-list-of-channels))]]))
+                mock-list-of-channels)])
+
+(def channel-list-component (memoize channel-list-component-fn))
+
+(defn render-page-content [{:keys [name description locked joined
+                                   status tokens tags community-color] :as community} channel-heights first-channel-height]
+  (fn [scroll-height]
+    [rn/view {:flex 1
+              :border-radius (scroll-page/diff-with-max-min scroll-height 16 0)
+              :background-color (colors/theme-colors
+                                 colors/white
+                                 colors/neutral-90)}
+     [rn/view
+      [rn/view {:padding-horizontal 20}
+       [rn/view {:border-radius    40
+                 :border-width     1
+                 :border-color     colors/white
+                 :position         :absolute
+                 :top              (if (<= scroll-height scroll-page/scroll-0)
```

Comment:
Since you're refactoring, I think all these calculations are too low level for a higher component such as overview. It makes reading the rest of the code harder. It'd be okay if they were present in the component library quo2, but not here. Perhaps if you extract some things into a `style` namespace it would be enough.

Reply:
yep, done